android/src/main/java/com/rivereactnative/RiveReactNativeView.kt

```diff
@@ -108,6 +114,44 @@ class RiveReactNativeView(private val context: ThemedReactContext) : FrameLayout
     addView(riveAnimationView)
   }
 
+  @OptIn(ControllerStateManagement::class)
+  override fun onAttachedToWindow() {
+    // The below solves: https://github.com/rive-app/rive-react-native/issues/198
+    // When the view is returned to, we reuse the view's resources.
+    // For example, navigating to a new page and returning, or a TabView.
+    controllerState?.let {
+      riveAnimationView.restoreControllerState(it);
+      // The controller refCount is a combination of the creation of the controller and the
+      // initialization of the RiveArtboardRenderer (which calls acquire on the controller).
+
+      // This could be decoupled in future versions of the Android runtime, and this code
+      // may require updating.
+      //
+      // For now we're doing a safety check to only add an additional acquire if the refCount is 0.
+      // As the RiveFileController would automatically have incremented the refCount on instantiation,
+      // and decreased when navigating away. It's safe to assume it should not be 0 at this point as
+      // the view still exists.
+      if (riveAnimationView.controller.refCount == 0) {
+        riveAnimationView.controller.acquire();
```

Comment:
Just taking a peek at this, I'm not entirely sure this is a good idea. If a controller reached 0, then all the resources associated with it have been released (i.e. `File`, `Artboard`, etc.). So if you restore a deleted controller, you're basically handing over broken resources to the View.

Reply:
Hey @HayesGordon and @zplata, this bit here is still problematic.
If this case is verified, the View will crash: once `refCount` reached 0, all the resources associated with it will be released

Reply:
@umberto-sonnino  That doesn't appear to be the case though. With the sample provided the view works fine, also for the case where you navigate to a new screen and back

Reply:
Mmh, when is this branch being called? 
Controllers are initialized with a `refCount` of 1, so it won't be called on first run.
When you navigate away (i.e. `onDetachedFromWindow`) this calls `saveControllerState()`, which ups the count by 1.
Does the controller ever reach 0? When will it be released?